nano/node/messages.cpp

```diff
@@ -1287,6 +1381,11 @@ bool nano::telemetry_ack::is_empty_payload () const
 	return size () == 0;
 }
 
+std::string nano::telemetry_ack::to_string () const
+{
+	return data.to_string ();
```

Comment:
It is possible for telemetry data to have empty payload. You need to check is_empty_payload.